tests/modules/images/dominant-color/dominant-color-test.php

```diff
@@ -108,6 +108,25 @@ public function test_tag_add_adjust_to_image_attributes( $image_path, $expected_
 		$this->assertEquals( $filtered_image_mock_lazy_load, $filtered_image_tags_not_added );
 	}
 
+	/**
+	 * Tests dominant_color_img_tag_add_dominant_color() do not override the style attribute.
+	 *
+	 * @covers ::dominant_color_img_tag_add_dominant_color
+	 */
+	public function test_dominant_color_do_not_override_the_style_attribute_in_img_tag() {
+		$attachment_id = self::factory()->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dominant-color/red.jpg' );
+		wp_maybe_generate_attachment_metadata( get_post( $attachment_id ) );
+
+		list( $src, $width, $height ) = wp_get_attachment_image_src( $attachment_id );
+
+		$filtered_image = sprintf( '<img src="%s" width="%d" height="%d" />', $src, $width, $height );
+
+		$this->assertStringContainsString( 'style="--dominant-color: #fe0000;"', dominant_color_img_tag_add_dominant_color( $filtered_image, 'the_content', $attachment_id ) );
+
+		$filtered_image_with_style_tag = sprintf( '<img style="color: #ffffff;" src="%s" width="%d" height="%d" />', $src, $width, $height );
+
+		$this->assertStringContainsString( 'style="--dominant-color: #fe0000; color: #ffffff;"', dominant_color_img_tag_add_dominant_color( $filtered_image_with_style_tag, 'the_content', $attachment_id ) );
```

Comment:
The first assertion appears to be testing the expected behaviour of `dominant_color_img_tag_add_dominant_color()` when there are no existing inline styles.

Should this test method actually be two different test methods?

- `test_dominant_color_img_tag_add_dominant_color_should_add_dominant_color_inline_style()`
- `test_dominant_color_img_tag_add_dominant_color_should_not_replace_existing_inline_styles()`

Reply:
Alternatively, you could use a data provider, passing `$filtered_image` and `$expected` to the test method.

The test method itself could be called `test_dominant_color_img_tag_add_dominant_color_should_add_dominant_color_inline_style()`.

With the datasets:
```php
return array(
	'no existing inline styles' => array(
		'filtered_image' => '<img src="%s" width="%d" height="%d" />',
		'expected'       => 'style="--dominant-color: #fe0000;"',
	),
	'existing inline styles'    => array(
		'filtered_image' => '<img style="color: #ffffff; style="%s" width="%d" height="%d" />',
		'expected'       => 'style="--dominant-color: #fe0000; color: #ffffff;"',
	),
);
```

Reply:
Thank you Colin for the feedback. Unit test updated in [1dc7a4e](https://github.com/WordPress/performance/pull/582/commits/1dc7a4e84a416e77c0fb171d1be2cc1603d1e5f7)